File: 프로그래머스/unrated/120861. 캐릭터의 좌표/캐릭터의 좌표.js
function solution(keyinput, board) {
    const [width, height] = board;
    const centerX = Math.floor(width / 2);
    const centerY = Math.floor(height / 2);
    let curX = 0, curY = 0;

    for (const key of keyinput) {
        if (key === 'left' && curX > -centerX) {
            curX--;
        } else if (key === 'right' && curX < centerX) {
            curX++;
        } else if (key === 'up' && curY < centerY) {
            curY++;
        } else if (key === 'down' && curY > -centerY) {
            curY--;
        }
    }

    return [curX, curY];
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
import java.util.*;

class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] answer = new int[2]; 
        int maxX = board[0] / 2;
        int maxY = board[1] / 2;

        for (String s : keyinput) {
            move(answer, s, maxX, maxY);
        }

        return answer;
    }

    private void move(int[] position, String direction, int maxX, int maxY) {
        switch (direction) {
            case "right" -> {
                if (position[0] < maxX) position[0]++;
            }
            case "left" -> {
                if (position[0] > -maxX) position[0]--;
            }
            case "up" -> {
                if (position[1] < maxY) position[1]++;
            }
            case "down" -> {
                if (position[1] > -maxY) position[1]--;
            }
        }
    }
}
